src/student/commands/list-student-schedules.command.ts

```diff
@@ -58,9 +63,13 @@ export class ListStudentSchedulesCommand {
     if (!schedules)
       throw new NotFoundException('Agendamentos nao encontrados.');
 
-    schedules.forEach((element) => {
-      if (element.monitor_id == monitor?.id) element['is_monitoring'] = true;
-      else element['is_monitoring'] = false;
+    schedules.forEach((schedule) => {
+      const monitoring = monitorings.find(
+        (monitor) => monitor?.id === schedule.monitor_id,
+      );
+      if (monitoring && monitoring.id_status == MonitorStatus.AVAILABLE)
+        schedule['is_monitoring'] = true;
+      else schedule['is_monitoring'] = false;
```

Comment:
Fica melhor assim:

```suggestion
    schedules.forEach((schedule) => {
      if (schedule.monitor.student_id === userId) schedule['is_monitoring'] = true;
      else schedule['is_monitoring'] = false;
```

Reply:
Como você já sabe que os agendamentos que vierem tem a participação do usuário logado de alguma forma, se vc verificar se o monitor do agendamento é o usuário logado já é o suficiente pra saber se é um agendamento em que ele foi monitor ou aluno. Se não foi monitor, só pode ter sido aluno e vice-versa.

Reply:
Outra coisa é que o is_monitoring tem que ser verdadeiro sempre que o usuário logado for o monitor do agendamento, independentemente do status da monitoria.